pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -571,12 +571,15 @@ public SuccessResponse reloadAllSegmentsDeprecated2(
   @ApiOperation(value = "Delete a segment", notes = "Delete a segment")
   public SuccessResponse deleteSegment(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
-      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") @Encoded String segmentName) {
+      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
+      @ApiParam(value = "Whether to delete the segment instantly or move to deleted_segment prefix and let "
+          + "RetentionManager handle the actual file deletion") @QueryParam("instantDelete") @DefaultValue("false")
```

Comment:
```suggestion
          ) @QueryParam("deleteWithoutBackup") @DefaultValue("false")
```

Reply:
with https://github.com/apache/pinot/pull/8176 we will be able to change the API to 
```suggestion
          + @ApiParam(value = "segment delete retention period, (for example 12h, 3d); setting it to 0d will instant delete the segments") @QueryParam("deleteRetention") @DefaultValue("7d") deleteRetentionPeriod
```